### ISSUE TITLE
[FP] RedundantAccessToObject срабатывает для внеконтекстных методов

### ISSUE THREAD

**Issue by vkostyanetsky** [1c-syntax/bsl-language-server#1918]
## Диагностика
RedundantAccessToObject

## Версия
Плагин 1.10.0 (0.19.0)

## Описание ложного срабатывания диагностики
Диагностика срабатывает при использовании конструкций вида ThisObject.AttributeName в модуле формы внутри внеконтекстных методов, для которых ThisObject — атрибут метода (то есть, обращение через точку не является избыточным).

## Пример кода
```
&AtClientAtServerNoContext
Procedure SetVisibilityAndAccessibilityOfAttributes(ThisObject, Reason = Undefined)
	
	Items = ThisObject.Items;
```

## Скриншоты
![firefox_UX7aFuCrN3](https://user-images.githubusercontent.com/40140090/143030856-e1a4ff6d-5242-47bb-a546-09d2c3172422.png)

**Comment by nixel2007** [comment]
Кажется, это было исправлено в 0.19.0

**Comment by asosnoviy** [comment]
> RedundantAccessToObject

С ходу не вижу фикса и в тестах тоже

**Comment by vkostyanetsky** [comment]
Так, проверил и понял, что ошибся с актуальной версией ПО — у нас сейчас стоит v1.10.0 (0.19.0). Поправил в тексте выше. 

Так что, видимо, в 0.19.0 эта проблема не исправлялась.

**Comment by nixel2007** [comment]
Ок, проверим, спасибо!

**Comment by asosnoviy** [comment]
если честно, я бы эту диагностику на две распилил и на ThisObject не орал бы по умолчанию.
